### DISCUSSION TITLE
Edrumulus for taiko drumming?

### DISCUSSION THREAD

**Discussion by m-k-S** [corrados/edrumulus#77] [General]
Hello,

I am a taiko drummer (Japanese drums, maybe you have seen them like here: https://www.youtube.com/watch?v=UajKUdALiLE) and I'd like to build my own taiko drum for practicing at home – but they are extremely loud, so electric seems like a better option. However, the only manufacturer of an electric taiko is Roland, and their model is $1400 which is not very affordable.

I am a software engineer with some familiarity with electrical engineering but no direct experience in audio / digital signal processing. Please forgive me ignorance, but would anyone mind answering a few questions I have? Specifically:

1. From my understanding - this is an interface between commercial electronic drum pads and a MIDI processor or DAW. Is that correct? In other words, to get fully up and running, all I need is a drum pad with a piezo trigger, and a computer running a DAW?

2a. In taiko an essential part of the technique is playing on different parts of the head to generate different pitch & tone, as well as on the rims. For this, my understanding is that I will need to attach several piezos both under the drumhead and along the rim to record my hits. Is that correct? 

2b. I read the discussion on this and saw that there are some concerns about position detection causing additional latency on the ESP32. Does anyone think that the Teensy 4.1 with its higher clock speed might fare better?

3. Are the piezo triggers in e.g. a Roland drumpad pre-amplified at all? Or is it just a bare piezo wired directly to a 6.35mm or 3.5mm output jack?

4. As far as re-creating the rich tone of the taiko is concerned, am I correct in understanding that this can be handled entirely on the DAW side, i.e. going into a studio and producing a high-quality MIDI instrument based on recordings?

Thanks so much in advance!

**Comment by corrados** [comment]
> I am a taiko drummer [...] and I'd like to build my own taiko drum for practicing at home

They have massive drum sticks and hit the instrument very hard. I could imagine that you drive the input in the clipping very easily. Maybe it would be required to change the values of the resistors for this application...

> From my understanding - this is an interface between commercial electronic drum pads and a MIDI processor or DAW

That is correct. But you do not need a commercial electronic drum pad. I have not tried it myself but DIY pads will most likely also work.

> For this, my understanding is that I will need to attach several piezos both under the drumhead and along the rim to record my hits. Is that correct?

On Roland mesh head pads, you have a center mounted piezo which is enough to support positional sensing. The Roland TAIKO-1 seems also only have two piezos (head/rim): "_[Thanks to dual-zone trigger technology with positional sensing ...](https://www.roland.com/us/products/taiko-1/)_"

>  I read the discussion on this and saw that there are some concerns about position detection causing additional latency on the ESP32. Does anyone think that the Teensy 4.1 with its higher clock speed might fare better.

No, the latency on the ESP32 and Teensy 4.0 will be comparable.

> Are the piezo triggers in e.g. a Roland drumpad pre-amplified at all? Or is it just a bare piezo wired directly to a 6.35mm or 3.5mm output jack?

It's usually just a bare piezo. No pre-amplification in these pads.

> As far as re-creating the rich tone of the taiko is concerned, am I correct in understanding that this can be handled entirely on the DAW side, i.e. going into a studio and producing a high-quality MIDI instrument based on recordings?

Yes, this is correct.

**Reply by m-k-S** [reply]
Thank you for the response! 

I will have some Edrumulus PCBs fabricated on PCBway soon and will attempt to construct my own drum pad/triggers as well. I will also get familiar with the positional sensing code and see if I can make any contributions.

Looking forward to joining this project!
